lib/saml_idp/xml_security.rb

```diff
@@ -97,10 +95,10 @@ def signature_algorithm(options)
         if options[:get_params] && options[:get_params][:SigAlg]
           algorithm(options[:get_params][:SigAlg])
         else
-          ref_elem = REXML::XPath.first(self, '//ds:Reference', { 'ds' => DSIG })
+          ref_elem = document.at_xpath('//ds:Reference | //Reference', DS_NS)
```

Comment:
Probable bug: `at_xpath('//Reference', DS_NS)` only searches for nodes without any namespace declaration, but doesn't search nodes where the namespace is declared as the _default_ namespace.

Reply:
hm, i had initially used wildcards (`:*//Reference`) rather than the OR clause, but i decided since the original code ONLY found either no namespace or ds, i didn't want to change it (and that felt way too permissible), since i didn't know what impact that would have to existing integrations. so `'//ds:Reference | //Reference'` finds either the ds namespace if it exists, or no namespace, but that's it. i think we should definitely think about adding the ability to have different namespaces, but that would be a feature, not a refactor.

with that extra context, does it seem like this will work as expected? thanks for helping me think through this!

Reply:
I think it makes sense to address this in a different PR as it is working, but may be more permissive that we'd like.

Reply:
we can make a ticket to define how we want namespace handling to work (ie, do we want to allow namespaces beyond `ds`) and refine from there!